src/components/sections/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,82 @@
+import type { ComponentPropsWithRef, FormEvent, ReactNode } from 'react'
+import React, { forwardRef, useRef } from 'react'
+import { Form, Label, Input, Button } from '@faststore/ui'
+
+export interface NewsletterProps
+  extends Omit<ComponentPropsWithRef<'form'>, 'onSubmit'> {
+  /**
+   * Heading for the section.
+   */
+  heading: ReactNode
+  /**
+   * Description about the newsletter.
+   */
+  description?: ReactNode
+  /**
+   * Callback function when submitted.
+   */
+  onSubmit: (value: string) => void
+}
+
+const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
+  function Newsletter({ heading, description, onSubmit, ...otherProps }, ref) {
+    const emailInputRef = useRef<HTMLInputElement>(null)
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+
+      if (emailInputRef.current?.value !== '') {
+        onSubmit(emailInputRef.current!.value)
+      }
+    }
+
+    return (
+      <Form
+        data-store-newsletter
+        ref={ref}
+        onSubmit={handleSubmit}
+        {...otherProps}
+      >
+        <div data-newsletter-content>
+          {heading}
+          {Boolean(description) && description}
+        </div>
+
+        <div data-newsletter-controls>
+          <Label htmlFor="newsletter-email">Your email</Label>
+          <Input
+            id="newsletter-email"
+            type="email"
+            name="newsletter-email"
+            ref={emailInputRef}
+          />
+          <Button type="submit">Subscribe</Button>
```

Comment:
I notice some variations on the Figma prototypes with more inputs. Should we implement it afterward? or maybe pass this piece as `children` prop?

Reply:
I tried to keep this one simple, so let's implement the more complex scenario afterward. 

I don't know yet how to best handle the input values inside `onSubmit` if we allow a variable number of inputs via `children`. 
😕